include/hipSYCL/runtime/dag_node.hpp

```diff
@@ -93,6 +106,16 @@ class dag_node
 
   void for_each_nonvirtual_requirement(std::function<void(dag_node_ptr)>
                                            handler) const;
+  /// Iterates across all operations that have actually been executed.
+  /// Precondition: is_submitted() returns true
+  template<class Handler>
+  void for_each_executed_operation(Handler h) {
```

Comment:
This function is added but not currently used, is that intentional?

Reply:
It provides a mechanism to access instrumented requirements. It's not yet used because it's unclear what should be returned for user-facing queries in case a requirement was optimized away, but I have some other PRs coming up that use that use it for internal runtime functionality.